common/oscanner.cpp

```diff
@@ -326,4 +332,146 @@ bool OScanner::isQuotedString() const
 	return _isQuotedString;
 }
 
+//
+// Get token as an int.
+//
+int OScanner::getTokenAsInt() const
+{
+	// fix for parser reading in commas
+	std::string str = _token;
+
+	// remove comma if necessary
+	if (str[str.length() - 1] == ',')
+	{
+		str[str.length() - 1] = '\0';
+	}
+
+	char* stopper;
+
+	if (str == "MAXINT")
+	{
+		return MAXINT;//INT32_MAX;
+	}
+
+	const int num = strtol(str.c_str(), &stopper, 0);
+
+	if (*stopper != 0)
+	{
+		I_Error("Bad numeric constant \"%s\".", str.c_str());
+	}
+
+	return num;
+}
+
+//
+// Get token as a float.
+//
+float OScanner::getTokenAsFloat() const
+{
+	// fix for parser reading in commas
+	std::string str = _token;
+
+	// remove comma if necessary
+	if (str[str.length() - 1] == ',')
+	{
+		str[str.length() - 1] = '\0';
+	}
+
+	char* stopper;
+
+	const double num = strtod(str.c_str(), &stopper);
+
+	if (*stopper != 0)
+	{
+		I_Error("Bad numeric constant \"%s\".", str.c_str());
+	}
+
+	return static_cast<float>(num);
+}
+
+//
+// Get token as a bool.
+//
+bool OScanner::getTokenAsBool() const
+{
+	return stricmp(_token.c_str(), "true") == 0;
+}
+
+//
+// Ensure next token is a string.
+//
+void OScanner::mustGetString()
+{
+	if (!scan())
+	{
+		error("Missing string (unexpected end of file).");
+	}
+}
+
+//
+// Ensure next token is an int.
+//
+void OScanner::mustGetInt()
+{
+	if (!scan())
+	{
+		error("Missing integer (unexpected end of file).");
+	}
+
+	// fix for parser reading in commas
+	std::string str = _token;
+
+	// remove comma if necessary
+	if (str[str.length() - 1] == ',')
+	{
+		str[str.length() - 1] = '\0';
+	}
+
+	if (IsNum(str.c_str()) == false || str != "MAXINT")
+	{
+		error("Missing integer (unexpected end of file).");
+	}
+}
+
+//
+// Ensure next token is a float.
+//
+void OScanner::mustGetFloat()
+{
+	if (!scan())
+	{
+		error("Missing floating-point number (unexpected end of file).");
+	}
+
+	// fix for parser reading in commas
+	std::string str = _token;
+
+	// remove comma if necessary
+	if (str[str.length() - 1] == ',')
+	{
+		str[str.length() - 1] = '\0';
+	}
+
+	if (IsRealNum(str.c_str()) == false)
+	{
+		error("Missing floating-point number (unexpected end of file).");
+	}
+}
+
+//
+// Ensure next token is a bool.
+//
+void OScanner::mustGetBool()
+{
+	if (!scan())
+	{
+		error("Missing boolean (unexpected end of file).");
+	}
+	
+	if (stricmp(_token.c_str(), "true") != 0 && stricmp(_token.c_str(), "false") != 0)
```

Comment:
We have a function specifically for dealing with std::string case-insensitive comparisons:

https://github.com/odamex/odamex/blob/stable/common/cmdlib.cpp#L240